### PR TITLE
feat: add support for rendering an optional PrometheusOperator ServiceMonitor resource definition

### LINKED ISSUE #227: Chart[Network Node]: Add support for rendering an optional PrometheusOperator ServiceMonitor resource definition
## User Story

**As an engineer,**

**I need**
> to capture metrics from the node software and other components which support a Hedera network deployment.

**So that**
> I can reliably determine the operational state of the network and make a pass/fail determination for a given test.

## Requirements

- Implement an optional `ServiceMonitor` definition for the network node container gated by a boolean flag defined as a chart value. 
   - https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/running-exporters.md#servicemonitors
- There should only be a single `ServiceMonitor` which handles all network node pods using a label selector.

### PR DESCRIPTION
**Pull request by leninmehedy**
## Description

This pull request changes the following:

- Adds support for deploying prometheus service monitor for all network nodes
- Adds helper scripts to deploy prometheus locally
- Update CI/CD pipeline to deploy prometheus-operator
- Update dev/Makefile to install prometheus-operator (if it does not exist) before deploying the network locally

### Notes:
- It was tested using prometheus operator: v0.67.1

### Related Issues

- Closes #227

### CODE REVIEW

File: charts/hedera-network/templates/services/network-node-svc.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ apiVersion: v1
 kind: Service
 metadata:
   name: network-{{ $nodeConfig.name }}-svc
+  labels:
+    fullstack.hedera.com/type: network-node-svc
```
Comment (nathanklick) [inline charts/hedera-network/templates/services/network-node-svc.yaml:8]: We should probably add a node name label here for use with label selector specs.

```suggestion
    fullstack.hedera.com/type: network-node-svc
    fullstack.hedera.com/node-name: {{ $nodeConfig.name }} 
```

File: charts/hedera-network/templates/telemetry/prometheus-svc-monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: network-node-monitor
+  labels:
+    fullstack.hedera.com/type: network-node-svc-monitor
+spec:
+  selector:
+    matchLabels:
+      fullstack.hedera.com/type: network-node-svc
+  endpoints:
+  {{- toYaml $.Values.telemetry.prometheus.svcMonitor.endpoints | nindent 4 }}
```
Comment (nathanklick) [inline charts/hedera-network/templates/telemetry/prometheus-svc-monitor.yaml:13]: Should this really be a values file definition? I do not believe the service definition itself is configurable, right?

File: charts/hedera-network/values.yaml [hunk 3]
```diff
@@ -10,6 +10,15 @@ cloud:
...
 
+# telemetry configurations
+telemetry:
+  prometheus:
+    svcMonitor:
+      enable: "true"
+      endpoints:
+        - port: metrics
+          interval: 5s
+
```
Comment (nathanklick) [inline charts/hedera-network/values.yaml:21]: Should this really be a values file entry? Wouldn't it make more sense as a partial template? 

What I am referring to is this portion: 

```
endpoints:
        - port: metrics
          interval: 5s
```

File: charts/hedera-network/templates/telemetry/prometheus-svc-monitor.yaml (hunk 2 above)
Comment (leninmehedy) [reply charts/hedera-network/templates/telemetry/prometheus-svc-monitor.yaml]: Removed the OTel port configurations from values file and updated the service definition

File: charts/hedera-network/values.yaml (hunk 3 above)
Comment (leninmehedy) [reply charts/hedera-network/values.yaml]: Removed endpoints from value file and added into the prometheus-svc-monitor

**Review by nathanklick — APPROVED** [review]

**Pull request merged**